src/server/game_server.rs

```diff
@@ -1,23 +1,38 @@
 use crate::actions::Action;
+use crate::entity::entity::EntityKind;
 use crate::network::server_update::ServerUpdate;
-use crate::network::server_update::ServerUpdate::{Attack, LoggedIn, RegisterEntity, SendAction, UpdatePosition};
+use crate::network::server_update::ServerUpdate::{Attack, LoggedIn, RegisterEntity, SendAction, UpdatePosition, RemoveEntity};
 use crate::primitives::position::Position;
+use crate::primitives::vector::Vector3;
 use crate::server::monster_manager::MonsterManager;
 use crate::server::server_state::ServerState;
 use crate::server::world_dispatcher::WorldDispatcher;
 use crate::world::World;
 use std::collections::HashMap;
 use std::sync::{Arc, Mutex};
-use std::time::Duration;
 use crate::attack::EntityAttack;
+use std::time::{Duration, Instant};
 
 /// Main function of the thread in charge of entities
 pub fn handle_entity_thread(server: Arc<Mutex<GameServer>>) {
-    let sleep_time = Duration::from_millis(20);
+    let sleep_time = Duration::from_millis(5);
 
+
+    let pos = Position::new(Vector3::new(0., 16., 0.), 0., 0.);
+    server.lock().unwrap().entity_server.spawn_new_monster(pos, EntityKind::Monster1);
+    let mut t = Instant::now();
+    let mut dt = 0.;
     loop {
-        server.lock().unwrap().entity_server.step(sleep_time);
-        std::thread::sleep(sleep_time)
+        dt = t.elapsed().as_secs_f32();
+        t = Instant::now();
+        
+        let player_list = server.lock().unwrap().state.connected_players().cloned().collect();
+        server.lock().unwrap().entity_server.step(dt, player_list);
+        server.lock().unwrap().update_buffers();
```

Comment:
Woa this is dangerous... The server will be always locked ! It's really not a good practice.